workspace_host/interface.js

```diff
@@ -46,14 +48,16 @@ app.use(bodyParser.json());
 
 // TODO: refactor into RESTful endpoints (https://github.com/PrairieLearn/PrairieLearn/pull/2841#discussion_r467245108)
 app.post('/', function(req, res) {
-    var workspace_id = req.body.workspace_id;
-    var action = req.body.action;
+    const workspace_id = req.body.workspace_id;
+    const state = req.body.state;
```

Comment:
Rather than sending the current state, let's send a parameter called `useInitialZip` that defaults to `false`.